hudi-flink/src/main/java/org/apache/hudi/sink/utils/Pipelines.java

```diff
@@ -189,21 +191,32 @@
   }
 
   public static DataStream<Object> hoodieStreamWrite(Configuration conf, int defaultParallelism, DataStream<HoodieRecord> dataStream) {
-    WriteOperatorFactory<HoodieRecord> operatorFactory = StreamWriteOperator.getFactory(conf);
-    return dataStream
```

Comment:
Can we have a separate method for hash index and not modify these two methods ? The methods are already too complex i think.

Reply:
man you forgot your previous comment lol. It used to be separated and you suggested to put them into `hoodieStreamWrite`. Let me know if I misunderstood.

Reply:
Yeah, if we can make both the code in `Pipelines` and `HoodieTableSink` clean ~